static/js/redux/actions/timetable_actions.jsx

```diff
@@ -148,6 +148,10 @@ export const lockTimetable = timetable => (dispatch, getState) => {
   }
 };
 
+const generateId = () => {
```

Comment:
why do we need this function?

Reply:
The reason that Custom Slots were buggy before was that they all had the same Id so this function creates unique Ids.

Reply:
so is the implementation of `generateCustomEventId()` wrong then? if that function is not generating unique ids I think you should just reimplement it instead of adding a new function since there are other places we use `generateCustomEventId()` that rely on the assumption that the generated IDs are unique (e.g. when we create new custom events). also then we would avoid having to generate ID functions